### PR TITLE
Fixes the  "Page Not Found - We are Sorry" after successful login and logout from a desktop app. Now shows simple message.

### LINKED ISSUE #9401: Error getting page when using KeycloakInstalled adapter
### Describe the bug

I use the keycloak installed adapter in a desktop java application. Previously, I would go through the login process, then the browser would show a screen that said 'Login Successful. You can close this page' while the application comes up in the background. Now, after logging in, the screen says 'Error: Page not found', though the application still opens successfully.

### Version

16.1.0

### Expected behavior

The browser should show that the login was successful.

### Actual behavior

The browser says redirects to a page that returns a 404.

### How to Reproduce?

The url I am redirected to (that previously said success and now says not found) is https://cerberus.mometrix.com/auth/realms/mometrix-internal/protocol/openid-connect/delegated
Navigating directly to this page without going through KeycloakInstalled gives the same result as going through KeycloakInstalled.

### PR DESCRIPTION
**Pull request by adame11**
Per @stianst comment on [PR 12796](https://github.com/keycloak/keycloak/pull/12796) "We need to fix KeycloakInstalled instead. The '/delegated' endpoint was a broken concept that should never have been introduced. It also has a CSRF issue as it's an open endpoint anyone can link to that will invalidate a users session. The proper fix here is to remove getRedirectUrl. Instead of redirecting it should show a message that the browser tab can be closed like it did previously". 

This PR attempts to do just that, without any fancy formatting, or customizability of the messages, which could be implemented in the future.

Closes #9401

### CODE REVIEW

**Comment by adame11** [comment]
I've added support for customized html and customized text only login success and logout for Keycloak installed adapter. It works by setting a Path to an optional static single HTML page to to server. Can only serve single page HTML (css and images are not served separately because webserver shuts down immediately after serving the single page). I also removed what seemed to be unused imports.

File: adapters/oidc/installed/src/main/java/org/keycloak/adapters/installed/KeycloakInstalled.java
Comment (stianst) [inline adapters/oidc/installed/src/main/java/org/keycloak/adapters/installed/KeycloakInstalled.java:72]:
```suggestion
    private static final String LOGGED_OUT = "logout";
```

Nit pick, but `loggedout` just doesn't feel right ;)

**Review by stianst — DISMISSED** [review]

**Comment by stianst** [comment]
A sensible solution to make KeycloakInstalled work again.

**Comment by adame11** [comment]
glad this meets the need. I accepted your "nitpick" wording change to use present tense logout instead of passed tense logged out (you're the boss, your call!! :) ). I wasn't clear on what to do next regarding the various CI/CD failed checks, let me know if there is something else.

**Comment by sennew** [comment]
Hi @stianst, is there a possibility to give me a timespan about when this fix would be released?

**Comment by adame11** [comment]
Hi @stianst, it's been a while... wondering if this can be approved (again) and merged?

**Comment by linm02** [comment]
Hello @stianst same question as above pretty please. It would help us to get this merged as well, is there a chance that this will make it to some upcoming release? Thanks!

**Comment by jptned** [comment]
@stianst Any chance this could be merged? Using the `KeycloakInstalled` method is still included in [the documentation](https://www.keycloak.org/docs/latest/securing_apps/#_installed_adapter), but results in an error page after logging in from desktop.

**Comment by edkennard** [comment]
Hello @stianst , we are also waiting for this fix, it's blocking us from upgrading to the Quarkus distribution which contains some improvements we'd love to have. Do you plan to approve and merge?

**Comment by HoaMai0205** [comment]
> Hello @stianst , we are also waiting for this fix, it's blocking us from upgrading to the Quarkus distribution which contains some improvements we'd love to have. Do you plan to approve and merge?

I'm not waiting for the merge. I go to "Files Changed" and get the raw file to create a custom like this:
image

**Comment by adame11** [comment]
> [HoaMai0205](/HoaMai0205)

Hi [HoaMai0205](https://github.com/HoaMai0205). I'm really sorry that I don't have an answer for you about why this has not been merged yet or when it will be. I'm not sure what's going on myself and can only hope that @stianst or others could chime in. In the meantime, I too have make my own custom copy of `KeycloakInstalledCustom` based off of KeycloakInstalled as a workaround.

**Comment by adame11** [comment]
> @adame11 Can you also share how-to create and use custom KeycloakInstalledCustom? It`s been a while waiting for this fix.

One way to utilize the fix in your own project while waiting for it to be merged is:

- copy the contents of https://github.com/keycloak/keycloak/blob/7ebdff99f63845b49742e75cb8dd2e344a58de64/adapters/oidc/installed/src/main/java/org/keycloak/adapters/installed/KeycloakInstalled.java
- Make a new Java class called `KeycloakInstalledCustom` 
- Use it as you would the original `KeycloakInstalled`
- customize login success and logout success messages for example:

```
//deployment param is outside scope of this example but gets created using something like deployment = KeycloakDeploymentBuilder.build(config);
KeycloakInstalledCustom keycloak = new KeycloakInstalledCustom(deployment); 
//optionally, set text-only output messages or true to include HTML tags
keycloak.setIsMessageHTML(false);
keycloak.setMessageLoginSuccess("Login successful. You may close this browser window and go back to NAME OF YOUR APPLICATION.");
keycloak.setMessageLogoutSuccess("Logged out. You are logged out of NAME OF YOUR APPLICATION. You may close this browser window.");
```
I'm sorry it's been a while. I'm not sure why this has not been merged in yet. See comments above.

**Comment by deamn** [comment]
Hi @stianst, I'm also interested, please review this PR

**Review by mposolda — APPROVED** [review]
@adame11 Approving, but I am not sure if/when `KeycloakInstalled` adapter is going to be deprecated or removed. But Thanks for the contribution!

**Review by pedroigor — APPROVED** [review]

**Comment by pedroigor** [comment]
@adame11 Sorry for the delay. Looks like it was already approved and I don't any other comment to add.

Can you please sign-off your commit so we can proceed with the merge?

**Comment by mposolda** [comment]
@adame11 @pedroigor It seems there is also compilation failure. Looks like rebase with latest Keycloak main would be needed to make things compilable again (in addition to the signature). Hope this helps.

**Comment by mposolda** [comment]
@adame11 Closing the PR due not progress in last weeks and doing cleanup of the PRs.

Moreover, I figured the plan is to remove `KeycloakInstalled` adapter in Keycloak 25 (together with other Keycloak OIDC and SAML adapters) as stated in https://www.keycloak.org/2024/03/keycloak-2400-released (the section "Java adapter deprecation and removal", which is about the removal of all java OIDC adapters).

If you are still interested in `KeycloakInstalled`, the option might be either to rely on older Keycloak release (not ideal) or move it to separate extension and maintain it yourself. The extension could then be possibly added to be listed in Keycloak extensions - https://www.keycloak.org/extensions.html

**Pull request closed without merging**
